cli/how-to-train-autorest.py

```diff
@@ -0,0 +1,160 @@
+    DatastorePropertiesResource,
+    InputDataBinding,
+    JobBaseResource,
+    Objective,
+    SweepJob,
+    TrialComponent,
+    DockerImage,
+    EnvironmentSpecificationVersion,
+    EnvironmentSpecificationVersionResource,
+)
+
+credentials, *_ = get_azure_cli_credentials()
+subscription_id = "6560575d-fa06-4e7d-95fb-f962e74efd7a"
+resource_group_name = os.environ["RG"]
+workspace_name = os.environ["WS"]
+
+client = AzureMachineLearningWorkspaces(credentials, subscription_id)
+print(
+    f"""Created AML client with CLI credentials for 
+        subscription: {subscription_id},
+        resource group: {resource_group_name},
+        workspace: {workspace_name}"""
+)
+
+# <create environment>
+conda_file = "name: python-ml-basic-cpu\nchannels:\n  - conda-forge\ndependencies:\n  - python=3.8\n  - pip\n  - pip:\n    - numpy\n    - pandas\n    - scipy\n    - scikit-learn\n    - matplotlib\n    - xgboost\n    - lightgbm\n    - dask\n    - distributed\n    - dask-ml\n    - adlfs\n    - fastparquet\n    - pyarrow\n    - mlflow\n    - azureml-mlflow"
+docker_spec = DockerImage(
+    docker_image_uri="mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04"
+)
+properties = EnvironmentSpecificationVersion(conda_file=conda_file, docker=docker_spec)
+request = EnvironmentSpecificationVersionResource(properties=properties)
+env_version = client.environment_specification_versions.create_or_update(
+    "lightgbm-environment",
+    1,
+    subscription_id=subscription_id,
+    workspace_name=workspace_name,
+    resource_group_name=resource_group_name,
+    body=request,
+)
+# </create environment>
+
+print(f"Created environment: {env_version}")
+
+# <create data>
+properties = DataVersion(
+    dataset_type="simple",
+    path="https://azuremlexamples.blob.core.windows.net/datasets/iris.csv",
+)
+request = DataVersionResource(properties=properties)
+data_version = client.data_versions.create_or_update(
+    "iris-data",
+    1,
+    body=request,
+    subscription_id=subscription_id,
+    workspace_name=workspace_name,
+    resource_group_name=resource_group_name,
+)
+# </create data>
+
+print(f"Created data version: {data_version}")
+
+# Create Code
+# TODO: decide how to upload to container
+
+datastores = client.datastores.list(
+    subscription_id=subscription_id,
+    workspace_name=workspace_name,
+    resource_group_name=resource_group_name,
+)
+datastore_id = list(filter(lambda d: d.name == "workspaceblobstore", datastores))[0].id
+print(f"Using datastore: {datastore_id}")
+
+# <create code>
+properties = CodeVersion(datastore_id=datastore_id, path="src")
+request = CodeVersionResource(properties=properties)
+code_version = client.code_versions.create_or_update(
+    "train-lightgbm",
+    1,
+    body=request,
+    subscription_id=subscription_id,
+    workspace_name=workspace_name,
+    resource_group_name=resource_group_name,
+)
+# </create code>
+
+print(f"Created code version: {code_version}")
+
+# TODO: figure out get compute
```

Comment:
we are adding compute support for this API version, after that we should be able to create compute autogen code as well